.pre-commit/check_header.py

```diff
@@ -46,6 +46,8 @@ def check_header(header_dir, files, verbose=False):
     for headertype in TYPES:
         with open(path.join(header_dir, headertype), 'r') as fhandle:
             header_content = fhandle.read()
+            if not hasattr(header_content, "decode"):
+                header_content = bytes(header_content, 'utf-8')
```

Comment:
ok, this was obviously my bad: I've implemented a regex against bytes with a str, sorry :(

Wouldn't it work to open the file as a bytes stream right from the beginning? So:
```python
with open(path.join(header_dir, headertype), 'rb') as fhandle:
```

Reply:
No worries, it was an easy catch.
You're right, what you suggest is a better solution, I just incorporated it.